### PR TITLE
Translucent fragments

### PR DESCRIPTION
**Pull request by Igor1201**
Keep old screen below the new one to be presented if the new screen's `screenColor` is translucent (to use like modals, for instance).

Supersedes https://github.com/airbnb/native-navigation/pull/60

### CODE REVIEW

**Comment by Igor1201** [comment]
c/c @gpeal @lelandrichardson

**Comment by Igor1201** [comment]
Actually, this PR won't work due to the addition of `ScreenCoordinatorLayout`. By controlling the drawing order there, when we keep the old view (don't `.detach()` it), the `reverseDrawing` check would be true and so the new view would be drawn below the old one.

**Comment by gusbicalho** [comment]
The logic on ScreenCoordinatorLayout to reverse the order of the drawing only works if it has exactly two children. If we do not detach the fragment, we'll end up with several children, and reversing the child draw order will not have the desired effect.

**Comment by gpeal** [comment]
@gusbicalho Yeah, reversing the last two views is probably correct for ScreenCoordinatorLayout. It has only been tested with the normal push/present case but could be modified to be more flexible.

**Comment by Igor1201** [comment]
@gpeal Our last commit seems to solve the issue we were discussing.

File: lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java [hunk 1]
```diff
@@ -100,6 +84,11 @@ public void onAnimationRepeat (Animation animation) {
   protected void dispatchDraw(Canvas canvas) {
     super.dispatchDraw(canvas);
 
+    if (drawingOps.size() < previousChildrenCount) {
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
```
Comment (Igor1201) [inline lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java]: @gpeal I guess we can keep track on children count on our side then ☝️

File: lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java [hunk 2]
```diff
@@ -43,12 +43,12 @@
  */
 public class ScreenCoordinatorLayout extends FrameLayout {
   private List<DrawingOp> drawingOpPool = new ArrayList<>();
-  private Stack<DrawingOp> drawingOps = new Stack<>();
-
-  private int previousBackStackEntryCount = 0;
-  private boolean reverseDrawing = false;
+  private List<DrawingOp> drawingOps = new ArrayList<>();
```
Comment (gpeal) [inline lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java]: These can both be final

File: lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java [hunk 3]
```diff
@@ -66,38 +66,54 @@ public void setFragmentManager(FragmentManager fragmentManager) {
...
     super.removeView(view);
   }
 
   @Override
   protected void dispatchDraw(Canvas canvas) {
     super.dispatchDraw(canvas);
 
-    while (!drawingOps.isEmpty()) {
-      DrawingOp op = drawingOps.remove(drawingOps.size() - 1);
+    if (drawingOps.size() < previousChildrenCount) {
```
Comment (gpeal) [inline lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java]: Can you add a comment here?

File: lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java [hunk 4]
```diff
@@ -66,38 +66,54 @@ public void setFragmentManager(FragmentManager fragmentManager) {
...
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
+    while (drawingOps.size() > 2) {
```
Comment (gpeal) [inline lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java]: 1) Maybe we can make a `drawAndRelease` helper
2) Can this loop just be in the else case and loop until !empty? Seems like all it's doing is finishing this loop. Then make the if case include this loop plus what's currently there.

File: lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java [hunk 5]
```diff
@@ -66,38 +67,52 @@ public void setFragmentManager(FragmentManager fragmentManager) {
...
   protected void dispatchDraw(Canvas canvas) {
     super.dispatchDraw(canvas);
 
+	// check the view removal is completed (by comparing the previous children count)
+    if (drawingOps.size() < previousChildrenCount) {
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
+    if (reverseLastTwoChildren) {
```
Comment (gpeal) [inline lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java]: @Igor1201 Would this be even more readable if you did an explicit swap of the last 2 items in the if block then just loop through them all?

**Comment by Igor1201** [comment]
Thanks for your patience! By the way, do we have a schedule of when to tag releases/push to npm?
